### PR TITLE
Fix for handling of legacy models that don't have a metadata file

### PR DESCRIPTION
**Pull request by fastier-li**

### CODE REVIEW

File: photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala [hunk 1]
```diff
@@ -510,28 +511,33 @@ object ModelProcessingUtils {
...
-      case _ => throw new RuntimeException(s"Couldn't find 'modelType' in metadata file: $inputPath")
-    }
+    val fs = Try(inputPath.getFileSystem(sparkContext.hadoopConfiguration))
+    val stream = fs.map(f => f.open(inputPath))
+    val fileContents = stream.map(s => Source.fromInputStream(s).getLines.mkString)
+
+    // TODO: log if we are in a legacy case and don't have metadata
+
+    params.taskType = fileContents.map {
+      fc => """"modelType"\s*:\s*"(.+?)"""".r findFirstMatchIn fc match {
```
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala]: What happened to `modelTypeRegularExpression`? Also, I forget the name of it, but that syntax is really discouraged. This is way more clear:
```
modelTypeRegularExpression.findFirstMatchIn(fc)
```

File: photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala [hunk 2]
```diff
@@ -510,28 +511,33 @@ object ModelProcessingUtils {
...
-   * @return A new copy of params with updates from the metadata file
+   * @return Either a new Param object, or Failure if a metadata file was not found, or it did not contain "modelType"
    */
   def loadGameModelMetadataFromHDFS(
-      sparkContext: SparkContext,
-      inputDir: String,
-      metadataFileName: String = "model-metadata.json"): Params = {
+    sparkContext: SparkContext,
+    inputDir: String,
+    metadataFileName: String = "model-metadata.json"): Params = {
```
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala]: The whitespace here was correct before, it should be 2 indents.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala (hunk 1 above)
Comment (fastier-li) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala]: It's been inlined... as for the syntax, that for you is more clear, I'll put back those dots :-)

File: photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala (hunk 2 above)
Comment (fastier-li) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala]: OK, I'll put that back.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala (hunk 1 above)
Comment (li-ashelkov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala]: https://github.com/databricks/scala-style-guide#infix

Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala]: https://github.com/databricks/scala-style-guide#infix

**Comment by fastier-li** [comment]
All comments from Alex addressed.

**Pull request merged**